task_manager_app/serializer.py

```diff
@@ -0,0 +1,64 @@
+from django.utils import timezone
+from rest_framework import serializers
+from rest_framework.response import Response
+
+from .models import Task, Category
+from .models import SubTask
+
+class SubTaskSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = SubTask
+        fields = '__all__'
+
+
+class TaskDetailSerializer(serializers.ModelSerializer):
+    subtasks = SubTaskSerializer(many=True, read_only=True)
+    class Meta:
+        model = Task
+        fields = '__all__'
+
+
+
+class CategoryCreateSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Category
+        fields = ['name']
+
+    def create(self, validated_data):
+        if Category.objects.filter(name=validated_data['name']).exists():
+            raise serializers.ValidationError({"name": "Category already exists"})
+        return super().create(validated_data)
+
+    def update(self, instance, validated_data):
+        if 'name' in validated_data and validated_data['name'] != instance.name:
+            if Category.objects.filter(name=validated_data['name']):
+                raise serializers.ValidationError({"name": "Category already exists"})
+            return super().update(instance, validated_data)
+
+
+class SubTaskCreateSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = SubTask
+        read_only_fields = ('created_at ',)
```

Comment:
Тут лишний пробел внутри строкового обозначения поля. Работать не будет
```suggestion
        read_only_fields = ('created_at',)
```